src/modules/sensors/vehicle_gps_position/gps_blending.cpp

```diff
@@ -584,6 +581,10 @@ void GpsBlending::calc_gps_blend_output(sensor_gps_s &gps_blended_state,
 
 			// sum weighted offsets
 			blended_alt_offset_mm += vert_offset * blend_weights[i];
+
+			// TODO: review
+			matrix::Vector3f temp_antenna_offset = _antenna_offset[i] * blend_weights[i];
+			antenna_offset += temp_antenna_offset;
```

Comment:
the problem for me with this implementation is that it doesn't take into account the filtering step: `_NE_pos_offset_m[i] = offset * alpha[i] + _NE_pos_offset_m[i] * (1.0f - alpha[i]);` (the blended position is not a weighted sum of the positions). Therefore the filtering is useless if we offset the antennas without filtering.
My idea would be to take the last blended position (after filtering) and compute the offset to the center of the drone for each GPS:
	1. For each GPS: compute the relative offset between GPS and blended output
	2. For each GPS: convert offset to the body frame
	3. For each GPS: compute blended antenna offset using antenna offset and relative pose between the GPS and blended output
	4. take a weighted average of the offsets of each GPS

Reply:
I think I would prefer to replace completely this blending logic by a sequential fusion of all the GNSS solutions, with a small 3D bias estimator for each source.

When a new GNSS position measurement arrives:
- compensate for distance to CG
- subtract estimated bias (we now have the measurement corrected for all known and estimated biases)
- update bias estimate with current position estimate and current GNSS measurement (improve the bias estimate using the fused solution of IMU and other aiding sources)
- update EKF2 position